src/Microsoft.AspNet.Cors.Core/CorsResult.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.AspNet.Cors.Core
+{
+    /// <summary>
+    /// Results returned by <see cref="ICorsEngine"/>.
+    /// </summary>
+    public class CorsResult
+    {
+        private long? _preflightMaxAge;
+
+        /// <summary>
+        /// Gets a value indicating whether the result is valid.
+        /// </summary>
+        public bool IsValid
+        {
+            get
+            {
+                return ErrorMessages.Count == 0;
+            }
+        }
+
+        /// <summary>
+        /// Gets the error messages.
+        /// </summary>
+        public IList<string> ErrorMessages { get; } = new List<string>();
```

Comment:
Sure .. I agree. The only reason I brought that up was I thought in the meeting we agreed to block the call. I will update the other two PRs ( MVC and the CORS Middleware) to reflect this.